Show CPU and memory usage in interactive mode

stroncium opened the pull request:
Fixes #62 .

Style and thresholds might be opinionated, but I tuned them to the point I was comfortable finding processes.

- processes are sorted by sum of cpu and memory usage(percentage), highest first
- in verbose mode memory and cpu usage is show when it is above 0.1% (minimal number we can get)
- in non-verbose mode memory usage is shown when it's above 0.5%, cpu usage shown when it's above 2%
- cpu usage is shown in dark red
- memory usage is shown in dark yellow

sindresorhus commented:
First thing I noticed is that it should not show `fkill` in the list.

I don't think they should use yellow and red color. Those are usually used to indicate high and medium usage. Are there any Unicode characters we could use to indicate memory and CPU instead of having to spell it out? Try to look for some.

Would be nice if the memory and CPU was updated in real-time (though of course not sort in real-time). The problem with only doing it at startup is that `iTerm` will always be close to the top as it's doing a lot of work startup `fkill-cli`.

stroncium commented:
I looked around for unicode symbols and found nothing even remotely similar. (Ignoring the fact they have high chance of not even being properly displayed in average terminal.)

Idea behind red and yellow was that CPU is highly limited, but memory is practically unlimited on modern systems(Which frequently have 16-32GB of RAM plus 64-128 GB of swap. To be honest it's been few years since I've encountered out of memory situation in real life, even when running really heavy DB queries on my notebook.). If you don't like that coloring, we can either chose neutral colors or just leave it uncolored, as it got MEM/CPU text indication already. In the end, I just forced some situations of me wanting to kill something and added indication which helped me to do it faster, might not be a universal thing. 
As for `iTerm` - I don't know what it is, but if it is putting so much stress on processor it feels fair to have it showed at top. We get percentage from `ps`, and `ps` measures average cpu use over process lifetime, not over some predetermined short amount of time.

sindresorhus commented:
> I looked around for unicode symbols and found nothing even remotely similar. (Ignoring the fact they have high chance of not even being properly displayed in average terminal.)

The reason I would prefer symbols is that MEM and CPU on each line looks very noisy. The symbols doesn't have to match exactly. We can document them at the top.

> If you don't like that coloring, we can either chose neutral colors or just leave it uncolored

Yeah, I would prefer it uncolored then.

> As for iTerm - I don't know what it is, but if it is putting so much stress on processor it feels fair to have it showed at top.

https://www.google.com/search?source=hp&ei=dIE7Xbz9DMuImwWQ94rwAQ&q=iterm&oq=iterm&gs_l=psy-ab.3..35i39j0i67l3j0l6.756.1241..1411...0.0..0.119.521.5j1......0....1..gws-wiz.....0.3KZudBFNliI&ved=0ahUKEwi8h77g09PjAhVLxKYKHZC7Ah4Q4dUDCAU&uact=5

stroncium commented:
@sindresorhus re: iTerm, so what is the point of not showing it if it uses so much CPU?

stroncium commented:
Honestly, I can't get anything even remotely similar. The best I've come up with is :memo: for memory and :gear: for cpu, but that's so arbitrary I can't make myself use em.

Note: I can't really test how it looks in actual terminal, takes a lot of work to set up terminal so it displays all the emojis without breaking normal text.

sindresorhus commented:
> The best I've come up with is 📝 for memory and ⚙️ for cpu, but that's so arbitrary I can't make myself use em.

Some suggestions.

CPU
- ⚡️
- 🚦
- 🚘 (speed/perf)

MEM
- 🍟(chips)
- 🐏(actually called "ram")
- 💭



> re: iTerm, so what is the point of not showing it if it uses so much CPU?

Because I'm running `fkill-cli` from iTerm, so the spike is actually launching `fkill-cli`, which is not very useful info as it's only a very short-lived spike.

stroncium commented:
Hmm, so, you launch new terminal and instantly execute fkill in it? Weird it get's such high CPU load, no matter what terminal I use the load produced is negligible.
Anyhow, I'm not quite sure I like doing such things(read: leaky abstractions), but if it really bothers you, I suppose we could just deprioritize iterm when sorting, but I need exact name of process.

sindresorhus commented:
> Hmm, so, you launch new terminal and instantly execute fkill in it?

No, I have the terminal running already.

sindresorhus commented:
Some minor things and we're done:

- `fkill 71167 ⚡64.4%` is showing. Same problem as iTerm showing. We should not show `fkill` at all.
- `Activity Monitor 67039 🐏1.0%` If it's just `.0%` at the end, remove the `.0` part.
- The lightning is not showing up as an emoji, so it looks uneven. It could also be confused with power consumption. Let's use `🚦` instead.
- The help text should document what 🚦 and 🐏 mean.

stroncium commented:
- Deprioritized `fkill` the same way as `iTerm`(for consistency).
 - Dropped `.0`s, but are you sure about that? Looked better(less messy) with `.0`s to me.
 - The way `⚡` and `🚦` are displayed depends on the list of fonts being used, so might be the other way around for different font settings. Switched to `🚦`.
 - Added note about emojis to help text.
 
Also:
 - Tweaked non-verbose ram/cpu thresholds a bit so it's more informative(0.5/2.0 => 1.0/3.0).
 - Aligned port-cpu-ram block to the right(when we know the screen width) to make it all less messy(especially in verbose mode), tell me what you think.

sindresorhus commented:
The port numbers are not highlighted when selected, so it's very hard to see which item they apply to.

<img width="1440" alt="Screenshot 2020-02-12 at 22 31 37" src="https://user-images.githubusercontent.com/170270/74349926-8dcc9300-4de7-11ea-9dfa-ad3a4f46ea3b.png">

sindresorhus commented:
I'm just gonna remove the magenta color for now, so this can be merged, but would be nice to properly fix this.

sindresorhus commented:
Thanks for working on this 👌
